core/inbox_test.go

```diff
@@ -31,20 +28,18 @@ func TestUpdateMessagePool(t *testing.T) {
 		// Msg pool: [m0, m1], Chain: b[]
 		// to
 		// Msg pool: [m0],     Chain: b[m1]
-		store, chainProvider := newStoreAndProvider(0)
+		builder, chainProvider := newBuilderAndProvider(t)
 		p := core.NewMessagePool(config.NewDefaultConfig().Mpool, th.NewMockMessagePoolValidator())
 		ib := core.NewInbox(p, 10, chainProvider)
 
 		m := types.NewSignedMsgs(2, mockSigner)
 		mustAdd(ib, m[0], m[1])
 
-		blk := types.Block{Height: 0}
-		parent := th.MustNewTipSet(&blk)
-
-		oldChain := core.NewChainWithMessages(store, parent, msgsSet{})
+		parent := builder.Genesis()
```

Comment:
This creates a *new* genesis block, distinct from the one created inside newBuilderAndProvider. You probably want `chainProvider.GetHead()` instead, to avoid confusion. Alternatively, inline the newBuilderAndProvider method.

Consider renaming `Genesis` to `NewGenesis`.